### DISCUSSION TITLE
Last update of opentelemetry-js deps broke OTLP exporter

### DISCUSSION THREAD

**Discussion by gizero** [renovatebot/renovate#25464] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

While trying to setup for the first time OTEL with renovate, I figured out that renovate releases with [this commit ](https://github.com/renovatebot/renovate/commit/2cc19e844) fail to export traces with the following error:

```
{
  traceId: 'c6d7585007301ea76006974313b586e9',
  parentId: undefined,
  traceState: undefined,
  name: 'run',
  id: '7c8bf044925e6a59',
  kind: 0,
  timestamp: 1698396138353000,
  duration: 27849956.199,
  attributes: {},
  status: { code: 0 },
  events: [],
  links: []
}
ERROR: unhandledRejection
       "err": {
         "name": "OTLPExporterError",
         "data": "{\"code\":3,\"message\":\"json: cannot unmarshal object into Go value of type uint64\"}",
         "code": 400,
         "message": "Bad Request",
         "stack": "OTLPExporterError: Bad Request\n    at IncomingMessage.<anonymous> (/opt/containerbase/tools/renovate/37.33.4/node_modules/@opentelemetry/otlp-exporter-base/src/platform/node/util.ts:131:27)\n    at /opt/containerbase/tools/renovate/37.33.4/node_modules/@opentelemetry/context-async-hooks/src/AbstractAsyncHooksContextManager.ts:75:49\n    at AsyncLocalStorage.run (node:async_hooks:327:14)\n    at AsyncLocalStorageContextManager.with (/opt/containerbase/tools/renovate/37.33.4/node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)\n    at IncomingMessage.contextWrapper (/opt/containerbase/tools/renovate/37.33.4/node_modules/@opentelemetry/context-async-hooks/src/AbstractAsyncHooksContextManager.ts:75:26)\n    at IncomingMessage.emit (node:events:529:35)\n    at IncomingMessage.emit (node:domain:489:12)\n    at endReadableNT (node:internal/streams/readable:1368:12)\n    at processTicksAndRejections (node:internal/process/task_queues:82:21)"
       }
```

Releases prior to the opentelemetry-js upgrade linked above are not affected. Looks like this is a [known issue](https://github.com/open-telemetry/opentelemetry-js/issues/4216). Looks like a fix has already landed to main branch but was not released yet. Other projects either discarded the upgrade or rolled back to the previous version. Given the pace Renovate releases new versions, it may be worth downgrading until after the upstream releases a new version with the fix. I can open a PR to downgrade the opentelemetry-js monorepo if that's the way to go. 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by HonkingGoose** [comment]
Upstream has a PR that fixes the OLTP JSON encoding problem:

- https://github.com/open-telemetry/opentelemetry-js/pull/4220

Looks like [the PR above is included in the `Experimental v0.45.0` release](https://github.com/open-telemetry/opentelemetry-js/releases/tag/experimental%2Fv0.45.0).

Renovate's own Dependency Dashboard shows a entry for `build(deps): update opentelemetry-js monorepo`. Maybe we should select the update and see if that fixes your problem?

**Reply by HonkingGoose** [reply]
Oh, looks like we already tried _something_ three weeks ago:

- #25503

I'll let the maintainers decide how to fix this then. :smile:

**Reply by gizero** [reply]
@HonkingGoose #25503 was before the upstream released the fix and it was required because renovate re-applied the broken update after it being rolled back. With the same PR we contextually introduced dashboard approval to avoid auto-merging the broken upgrade again. Now I locally tested the latest release from upstream and I can confirm that the original issue is resolved and traces are successfully sent to the OTLP collector again. I think we can remove the dashboard approval config snippet and wait for renovate to upgrade opentelemetry-js monorepo. I'll open a PR shortly today to address this.

**Reply by gizero** [reply]
Meanwhile, I'm ok with manually selecting the upgrade from the dashboard, of course.

**Reply by HonkingGoose** [reply]
I got a bit confused with the whole merge PR, revert, then new PR to revert properly again thing. So I didn't know what to do anymore. :smile: 

Thank you for testing the newest version. I'll let you create the PR to fix this then! :wink:

**Reply by gizero** [reply, marked as answer]
@HonkingGoose see https://github.com/renovatebot/renovate/pull/25888
